projects/ngx-avatar/src/lib/avatar.component.ts

```diff
@@ -253,20 +267,22 @@ export class AvatarComponent implements OnChanges, OnDestroy {
    * memberof AvatarComponent
    */
   private fetchAndProcessAsyncAvatar(source: AsyncSource): void {
+    if (this.avatarService.sourceHasFailedBefore(source)) {
+      return;
+    }
+
     this.avatarService
-      .fetchAvatar(source.getAvatar(this.size))
-      .pipe(
-        takeWhile(() => this.isAlive),
-        map(response => source.processResponse(response, this.size))
-      )
-      .subscribe(
-        avatarSrc => (this.avatarSrc = avatarSrc),
-        err => {
-          console.error(
-            `ngx-avatar: error while fetching ${source.sourceType} avatar `
```

Comment:
Wouldn't be nice to give some error in case no sources give success?

Reply:
Removing the console error was done by @HaithemMosbahi himself, in his [original PR](https://github.com/HaithemMosbahi/ngx-avatar/pull/59/files#diff-f6b322f18eb101b5701f89a8f88a0a9bR265). And I kinda agree with his choice, because this lib is error tolerant with a fallback mechanism in case of error to try the next source. I don't think it would be useful to dump lots of errors that are expected in the final application.

**If** we need to throw some kind of error, **maybe** we could do that when the entire fallback mechanism failed, and there is no possible source to try. That could be meaningful for the end-user, because he failed to provide at least one valid source, and he will see nothing at all instead of the expected avatar. But that is out of scope of this PR.

Reply:
You're right. Also, I went back to Haithem's PR some minutes after making this comment and saw it myself, it got me thinking. Instead of errors and stuff, we could just return to the user what source did the library pick in an `enum` cointaining the `Invalid` option or, instead of returning it, we could expose it via some output. But, of course, as you said that's out of the scope of this PR.